core/src/main/scala/kafka/tools/StorageTool.scala

```
@@ -260,6 +314,20 @@ object StorageTool extends Logging {
         s"${MetadataVersion.IBP_3_0_IV0}; the default is ${MetadataVersion.LATEST_PRODUCTION}")
   }
 
+  private def addFeatureDependenciesParser(subparsers: Subparsers): Unit = {
```

do we allow more than one feature flag to be provided? This is the case for the other commands, so wasn't sure the result here.

---

No we don't, this is ensured using store() which only allows one value. If multiple values are provided it takes the last one. Do we want to throw an exception if multiple features are listed or are we fine with the default behavior? `    featureDependenciesParser.addArgument("--feature", "-f")
      .required(true)
      .help("The feature and version to look up dependencies for, in feature=level format." +
        " For example: `metadata.version=5`."
      )
      .action(store())`

---

Would it be possible just to list them all?

---

yes on it